helix-core/src/main/java/org/apache/helix/task/assigner/ThreadCountBasedTaskAssigner.java

```diff
@@ -74,19 +75,16 @@ public Map<String, TaskAssignResult> assignTasks(Iterable<AssignableInstance> as
   public Map<String, TaskAssignResult> assignTasks(
       AssignableInstanceManager assignableInstanceManager, Collection<String> instances,
       Iterable<TaskConfig> tasks, String quotaType) {
-    Iterable<AssignableInstance> assignableInstances = new HashSet<>();
+    Set<AssignableInstance> assignableInstances = new HashSet<>();
     // Only add the AssignableInstances that are also in instances
     for (String instance : instances) {
-      ((HashSet<AssignableInstance>) assignableInstances)
-          .add(assignableInstanceManager.getAssignableInstance(instance));
+      assignableInstances.add(assignableInstanceManager.getAssignableInstance(instance));
```

Comment:
Looks like the below check of `assignableInstances` can be moved here - not that it helps with logs, but the logic can be simpliefied.

Reply:
You mean check on `assignableInstances.isEmpty()`? The logic will be wrong if move into the forloop. Am I missing anything?

Reply:
Synced offline: this is a NIT.